test/integration/model_photo_shoot.cc

```diff
@@ -0,0 +1,265 @@
+            scene->NodeByIndex(i));
+        if (nullptr != camera && camera->Name() == "photo_shoot::link::camera")
+        {
+          ignition::math::Pose3d pose;
+          // Perspective view
+          pose.Pos().Set(1.6 / scaling + translation.X(),
+                        -1.6 / scaling + translation.Y(),
+                        1.2 / scaling + translation.Z());
+          pose.Rot().Euler(0, IGN_DTOR(30), IGN_DTOR(-225));
+          SavePicture(camera, pose, "1_test.png");
+          // Top view
+          pose.Pos().Set(0 + translation.X(),
+                        0 + translation.Y(),
+                        2.2 / scaling + translation.Z());
+          pose.Rot().Euler(0, IGN_DTOR(90), 0);
+          SavePicture(camera, pose, "2_test.png");
+
+          // Front view
+          pose.Pos().Set(2.2 / scaling + translation.X(),
+                        0 + translation.Y(),
+                        0 + translation.Z());
+          pose.Rot().Euler(0, 0, IGN_DTOR(-180));
+          SavePicture(camera, pose, "3_test.png");
+
+          // Side view
+          pose.Pos().Set(0 + translation.X(),
+                        2.2 / scaling + translation.Y(),
+                        0 + translation.Z());
+          pose.Rot().Euler(0, 0, IGN_DTOR(-90));
+          SavePicture(camera, pose, "4_test.png");
+
+          // Back view
+          pose.Pos().Set(-2.2 / scaling + translation.X(),
+                        0 + translation.Y(),
+                        0 + translation.Z());
+          pose.Rot().Euler(0, 0, 0);
+          SavePicture(camera, pose, "5_test.png");
+        }
+      }
+      testPicturesTaken = true;
+    }
+  }
+
+  /// \brief Loads the pose values generated by the Model Photo Shoot plugin.
+  /// \param[in] _poseFile File containing the generated poses.
+  protected: void LoadPoseValues(std::string _poseFile = "poses.txt")
+  {
+    std::ifstream poseFile (_poseFile);
+    std::string line;
+    ASSERT_TRUE(poseFile.is_open());
+    while (getline(poseFile, line) )
+    {
+      std::istringstream iss(line);
+      std::string word;
+      while (getline( iss, word, ' ' ))
+      {
+        if (word == "Translation:")
+        {
+          float tr_x, tr_y, tr_z;
+          getline( iss, word, ' ' );
+          tr_x = std::stof(word);
+          getline( iss, word, ' ' );
+          tr_y = std::stof(word);
+          getline( iss, word, ' ' );
+          tr_z = std::stof(word);
+          this->translation = {tr_x, tr_y, tr_z};
+          break;
+        }
+        if (word == "Scaling:")
+        {
+          getline( iss, word, ' ' );
+          this->scaling = std::stod(word);
+          break;
+        }
+      }
+    }
+    poseFile.close();
+    EXPECT_EQ(remove(_poseFile.c_str()), 0);
+  }
+
+  // Documentation inherited
+  protected: void SetUp() override
+  {
+    InternalFixture::SetUp();
+
+    auto plugin = sm.LoadPlugin("libMockSystem.so",
```

Comment:
I recommend using the `TestFixture` class to avoid much of this boilerplate.

Reply:
I added it, looking much better, thanks!